src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -273,13 +256,14 @@ func (s *peersSource) readData(
 	var (
 		resultLock              sync.Mutex
 		wg                      sync.WaitGroup
-		persistenceWorkerDoneCh = make(chan struct{})
 		persistenceMaxQueueSize = s.opts.PersistenceMaxQueueSize()
 		persistenceQueue        = make(chan persistenceFlush, persistenceMaxQueueSize)
 		resultOpts              = s.opts.ResultOptions()
 		count                   = shardTimeRanges.Len()
 		concurrency             = s.opts.DefaultShardConcurrency()
 		blockSize               = nsMetadata.Options().RetentionOptions().BlockSize()
+		persistWg               = &sync.WaitGroup{}
```

Comment:
nit: sync.WaitGroup is more idiomatic here, yes?

Reply:
Oh, I get it. This is so you get the actual reference. Makes sense.

Reply:
you mean without the ref? since it's being passed around between functions you need the ref

Reply:
I think we're intending to say the same thing, but basically he's creating a pointer so the sync.WaitGroup object itself doesn't get copied when passed to the function using it.

Reply:
lol i think we commented at the same time